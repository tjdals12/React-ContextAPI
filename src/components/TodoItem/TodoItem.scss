.TodoItem {
    &.done {
        color: #999999;
    }

    .text {
        cursor: pointer;
    }

    .remove {
        cursor: pointer;
        color: red;
        margin-left: 10px;
    }
}
