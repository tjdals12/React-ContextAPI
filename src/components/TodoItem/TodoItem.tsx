import React from 'react';
import classNames from 'classnames/bind';
import styles from './TodoItem.scss';

const cx = classNames.bind(styles);

type TodoItemProps = {
    todo: { id: number; text: string; done: boolean };
};

function TodoItem({ todo }: TodoItemProps): JSX.Element {
    return (
        <li className={cx(`TodoItem ${todo.done ? 'done' : ''}`)}>
            <span className={cx('text')}>{todo.text}</span>
            <span className={cx('remove')}>(X)</span>
        </li>
    );
}

export default TodoItem;
