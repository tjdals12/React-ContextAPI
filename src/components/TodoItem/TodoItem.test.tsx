import React from 'react';
import { render } from '@testing-library/react';
import TodoItem from './TodoItem';

describe('<TodoItem />', () => {
    it('matches snapshot', () => {
        const utils = render(<TodoItem todo={{ id: 0, text: 'Something', done: false }} />);
        expect(utils.container).toMatchSnapshot();
    });

    it('has a text and a remove button', () => {
        const utils = render(<TodoItem todo={{ id: 0, text: 'Something', done: false }} />);
        expect(utils.container.querySelector('.text')).not.toBeNull();
        expect(utils.container.querySelector('.remove')).not.toBeNull();
    });
});
