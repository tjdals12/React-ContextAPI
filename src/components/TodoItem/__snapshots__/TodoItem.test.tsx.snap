// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> matches snapshot 1`] = `
<div>
  <li
    class="TodoItem "
  >
    <span
      class="text"
    >
      Something
    </span>
    <span
      class="remove"
    >
      (X)
    </span>
  </li>
</div>
`;
